include/sonic/internal/arch/avx2/base.h

```diff
@@ -157,6 +181,181 @@ sonic_force_inline void Xmemcpy<16>(void* dst_, const void* src_,
+  size_t addr = (size_t)(a) | (size_t)(b);
+  return ((addr) & (PageSize - 1)) <= (PageSize - VecLen);
+#endif
+}
+
+static sonic_force_inline int cmp_lt_32(const void* _l, const void* _r,
+                                        size_t s) {
+  auto lhs = static_cast<const uint8_t*>(_l);
+  auto rhs = static_cast<const uint8_t*>(_r);
+  if (in_page_32(lhs, rhs)) {
+    __m256i vec_l = _mm256_loadu_si256((__m256i const*)rhs);
+    __m256i vec_r = _mm256_loadu_si256((__m256i const*)lhs);
+    __m256i ans = _mm256_cmpeq_epi8(vec_l, vec_r);
+    int mask = _mm256_movemask_epi8(ans) + 1;
+    // mask = mask << (32 -s);
+    __asm__("bzhil  %1, %2, %[result]\n\t"
+            : [result] "=r"(mask)
+            : "r"((int)s), "r"(mask));
+    if (mask) {
+      int ne_idx = __builtin_ctz(mask);
+      // if (lhs[ne_idx] < rhs[ne_idx]) return -1;
+      // else return 1;
+      return lhs[ne_idx] - rhs[ne_idx];
+    } else {
+      return 0;
+    }
+  }
+  return std::memcmp(lhs, rhs, s);
+}
+
+// slow path
+static inline bool is_eq_lt_32_cross_page(const void* _a, const void* _b,
+                                          unsigned int s) {
+  auto a = static_cast<const uint8_t*>(_a);
+  auto b = static_cast<const uint8_t*>(_b);
+  if (s >= 16) {
+    __m128i vec_a = _mm_loadu_si128((__m128i const*)a);
+    __m128i vec_b = _mm_loadu_si128((__m128i const*)b);
+    __m128i ans1 = _mm_cmpeq_epi8(vec_a, vec_b);
+
+    vec_a = _mm_loadu_si128((__m128i const*)(a + s - 16));
+    vec_b = _mm_loadu_si128((__m128i const*)(b + s - 16));
+    __m128i ans2 = _mm_cmpeq_epi8(vec_a, vec_b);
+
+    __m128i ans = _mm_and_si128(ans1, ans2);
+    int mask = _mm_movemask_epi8(ans);
+    return mask == 0xFFFF;
+  }
+  // cross page
+  if (s >= 8) {
+    return __builtin_memcmp(a, b, 8) == 0 &&
+           __builtin_memcmp(a + s - 8, b + s - 8, 8) == 0;
+  } else if (s >= 4) {
+    return __builtin_memcmp(a, b, 4) == 0 &&
+           __builtin_memcmp(a + s - 4, b + s - 4, 4) == 0;
+  } else if (s >= 2) {
+    return __builtin_memcmp(a, b, 2) == 0 &&
+           __builtin_memcmp(a + s - 2, b + s - 2, 2) == 0;
+  } else {
+    return *a == *b;
+  }
+  return true;
+}
+
+static sonic_force_inline bool is_eq_lt_32(const void* _a, const void* _b,
+                                           size_t s) {
+  auto a = static_cast<const uint8_t*>(_a);
+  auto b = static_cast<const uint8_t*>(_b);
+  if (in_page_32(a, b)) {
+    __m256i vec_a = _mm256_loadu_si256((__m256i const*)a);
+    __m256i vec_b = _mm256_loadu_si256((__m256i const*)b);
+    __m256i ans = _mm256_cmpeq_epi8(vec_a, vec_b);
+    int mask = _mm256_movemask_epi8(ans) + 1;
+    // mask = mask << (32 -s);
+    __asm__("bzhil  %1, %2, %[result]\n\t"
+            : [result] "=r"(mask)
+            : "r"((int)s), "r"(mask));
+    return mask == 0;
+  }
+  return is_eq_lt_32_cross_page(a, b, s);
+}
+}  // namespace
+
+sonic_force_inline bool MemcmpEqShort(const void* _a, const void* _b,
```

Comment:
The ambiguous name means only doing `memcmpeq` for short strings. 

I suggest changing it to `InlinedMemcmpEq` or `FastMemcmpEq`

Reply:
Actually, It doesn't faster than std::memcmp when size is too large.